File: 2391-minimum-amount-of-time-to-collect-garbage/2391-minimum-amount-of-time-to-collect-garbage.py
class Solution(object):
    def garbageCollection(self, garbage, travel):
        garbage_type, ans, n, m = ['M', 'P', 'G'], 0, len(garbage), len(travel)
        
        for x in garbage_type:
            time, last_pick = 0, 0
            for i in range(n):
                count = 0
                for bin in garbage[i]:
                    if bin == x:
                        count += 1
                if count > 0:
                    time += count
                    last_pick = i
                if i < m:
                    time += travel[i]
            
            for j in range(n-2, last_pick-1, -1):
                time -= travel[j]
            
            ans += time
            
        return ans
                
                
        

File: 2391-minimum-amount-of-time-to-collect-garbage/2391-minimum-amount-of-time-to-collect-garbage.cpp
class Solution {
public:
    int garbageCollection(vector<string>& garbage, vector<int>& travel) {
        int ans = 0;
        vector<char> c = {'G', 'M', 'P'};
        for(char x : c){
            int time = 0, lastpick = 0;
            for(int i = 0; i<garbage.size(); i++){
                int cnt = count(garbage[i].begin(), garbage[i].end(), x);
                if(cnt > 0){
                    time += cnt;
                    lastpick = i;
                }
                if(i < travel.size()) time += travel[i];
            }
            for(int j = garbage.size() - 2; j>=lastpick; j--){
                time -= travel[j];
            }
            
            ans += time;
        }
        
        return ans;
    }
};